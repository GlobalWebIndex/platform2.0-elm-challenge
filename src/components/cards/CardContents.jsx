import React from "react";
import PropTypes from "prop-types";
import { useStyles } from "./Card.style";

const CardContents = ({
  imageSrc,
  imageWidth,
  imageHeight,
  id,
  title,
  children,
}) => {
  const classes = useStyles();
  return (
    <>
      {imageSrc && (
        <img
          className={classes.cardImage}
          src={imageSrc}
          alt={`id ${id}`}
          width={imageWidth}
          height={imageHeight}
          loading="lazy"
          data-testid="card-image"
        />
      )}
      {title && (
        <div data-testid="card-title" className={classes.cardTitle}>
          {title}
        </div>
      )}
      {children && <div className={classes.cardChildren}>{children}</div>}
    </>
  );
};

CardContents.propTypes = {
  imageSrc: PropTypes.string,
  imageHeight: PropTypes.number,
  imageWidth: PropTypes.number,
  id: PropTypes.string,
  title: PropTypes.string,
};

export default CardContents;
