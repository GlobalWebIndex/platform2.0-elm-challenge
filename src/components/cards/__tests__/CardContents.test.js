import CardContents from "../CardContents";
import { render, screen } from "@testing-library/react";

describe("<CardContents/>", () => {
  it("renders", () => {
    const { container } = render(
      <CardContents
        imageSrc="https://cat.image"
        imageHeight={311}
        imageWidth={111}
        id="abc"
        title="lovely cat"
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("renders a child component", () => {
    render(
      <CardContents title="lovely cats">
        <div data-testid="temp-component" />
      </CardContents>
    );

    expect(screen.getByTestId("temp-component")).toBeInTheDocument();
  });

  it("renders without title", () => {
    render(
      <CardContents
        imageSrc="https://cat.image"
        imageHeight={311}
        imageWidth={111}
        id="abc"
      />
    );

    expect(screen.queryByTestId("card-title")).not.toBeInTheDocument();
  });

  it("renders without an image", () => {
    render(<CardContents title="lovely cats" />);

    expect(screen.queryByTestId("card-image")).not.toBeInTheDocument();
  });
});
