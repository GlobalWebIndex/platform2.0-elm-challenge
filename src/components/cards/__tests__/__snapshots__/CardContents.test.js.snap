// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContents/> renders 1`] = `
<div>
  <img
    alt="id abc"
    class="cardImage-0-2-2"
    data-testid="card-image"
    height="311"
    loading="lazy"
    src="https://cat.image"
    width="111"
  />
  <div
    class="cardTitle-0-2-3"
    data-testid="card-title"
  >
    lovely cat
  </div>
</div>
`;
